crates/vision/src/line_detection.rs

```diff
@@ -82,19 +96,61 @@ impl LineDetection {
         let mut image_lines = Vec::new();
         let mut discarded_lines = Vec::new();
 
-        let LinePoints {
-            line_points,
-            used_segments,
-        } = filter_segments_for_lines(
-            context.camera_matrix,
-            context.filtered_segments,
-            context.image,
-            *context.check_line_segments_projection,
-            context.allowed_projected_segment_length,
-            *context.check_edge_gradient,
-            *context.gradient_alignment,
+        // let horizontal_scan_lines = iter_if(
+        //     *context.use_horizontal_segments,
+        //     context
+        //         .filtered_segments
+        //         .scan_grid
+        //         .horizontal_scan_lines
+        //         .iter(),
+        // );
+        let vertical_scan_lines = iter_if(
+            *context.use_vertical_segments,
+            context
+                .filtered_segments
+                .scan_grid
+                .vertical_scan_lines
+                .iter(),
+        );
+
+        // let horizontal_segments = map_segments::<HorizontalMapping>(
+        //     horizontal_scan_lines,
+        //     *context.maximum_merge_gap_in_pixels,
+        // );
+        let vertical_segments = map_segments::<VerticalMapping>(
+            vertical_scan_lines,
             *context.maximum_merge_gap_in_pixels,
         );
+
+        let filtered_segments = /*horizontal_segments
+            .chain(*/vertical_segments/*)*/
+            .filter(is_non_field_segment)
+            .filter(|segment| {
+                !*context.check_line_segments_projection
+                    || is_in_length_range(
+                        segment,
+                        context.camera_matrix,
+                        context.allowed_projected_segment_length,
+                    )
+            })
+            .filter(|segment| !*context.check_edge_gradient || is_maximum_segment(segment))
+            .filter(|segment| {
+                has_opposite_gradients(segment, context.image, *context.gradient_alignment, *context.gradient_sobel_stride)
+            });
+
+        let (line_points, used_segments): (Vec<Point2<Ground>>, HashSet<Point2<Pixel, u16>>) =
+            filtered_segments
+                .filter_map(|segment| {
+                    Some((
+                        context
+                            .camera_matrix
+                            .pixel_to_ground(segment.center().cast())
+                            .ok()?,
+                        segment.start,
```

Comment:
I think the `used_segments` feature as used by ball detection and feet detection is broken since we started merging segments in the line detection.

Reply:
It will break further when adding horizontal image segments since two segments in different directions may share the same start position, so a horizontal segment that passes the filters here may mask a vertical segment that might have been used (correctly) in the ball detection.

Reply:
Yes, that might be true. Do we want to fix it in this PR? I would like to get the large pieces in and integrate/fix them in main.